File: backend/app/schemas/review.py
from datetime import datetime

from pydantic import BaseModel, field_validator


class ReviewCreate(BaseModel):
    rating: int
    comment: str = ""

    @field_validator("rating")
    @classmethod
    def rating_in_range(cls, v: int) -> int:
        if not 1 <= v <= 5:
            raise ValueError("Rating must be between 1 and 5")
        return v


class ReviewOut(BaseModel):
    id: int
    course_id: int
    user_id: int
    reviewer_name: str
    rating: int
    comment: str
    created_at: datetime

File: mobile/lib/models/review.dart
class Review {
  final int id;
  final int courseId;
  final int userId;
  final String reviewerName;
  final int rating;
  final String comment;
  final DateTime createdAt;

  Review({
    required this.id,
    required this.courseId,
    required this.userId,
    required this.reviewerName,
    required this.rating,
    required this.comment,
    required this.createdAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) => Review(
    id: json['id'] as int,
    courseId: json['course_id'] as int,
    userId: json['user_id'] as int,
    reviewerName: json['reviewer_name'] as String,
    rating: json['rating'] as int,
    comment: json['comment'] as String,
    createdAt: DateTime.parse(json['created_at'] as String),
  );
}
